RIRImageResizing/Classes/NSObject/RIRResizeImageOperationParameters.m

```diff
@@ -0,0 +1,44 @@
+//
+//  RIRResizeImageOperationParameters.m
+//  Pods
+//
+//  Created by Richard Reitzfeld on 11/21/16.
+//
+//
+
+#import "RIRResizeImageOperationParameters.h"
+
+#import <ResplendentUtilities/RUConditionalReturn.h>
+
+
+
+
+
+@implementation RIRResizeImageOperationParameters
+
+#pragma mark - NSObject
+-(instancetype)init
+{
+    kRUConditionalReturn_ReturnValueNil(YES, YES);
+    return [self init_with_newSize:CGSizeZero
+                        resizeMode:UIImage_RIRResizing_ResizeMode_AspectFit];
+}
+
+#pragma mark - init
+-(instancetype)init_with_newSize:(CGSize)newSize
+                      resizeMode:(UIImage_RIRResizing_ResizeMode)resizeMode
+{
+    kRUConditionalReturn_ReturnValueNil(newSize.height < 0 || newSize.width < 0, NO);
+    kRUConditionalReturn_ReturnValueNil(resizeMode < UIImage_RIRResizing_ResizeMode_first || resizeMode > UIImage_RIRResizing_ResizeMode_last, NO);
```

Comment:
For testing if an enum is out of range, do the following:

1. Go to where you define the types.
2. Import `RUEnumIsInRangeSynthesization.h`
3. After you declare all the types, add the following line:
```
static inline RUEnumIsInRangeSynthesization_autoFirstLast(UIImage_RIRResizing_ResizeMode);
```

Which will generate the method for you `UIImage_RIRResizing_ResizeMode__isInRange()`. Use that to know if an enum is in range.